cmd/frontend/graphqlbackend/search.go

```diff
@@ -365,9 +366,45 @@ func exactlyOneRepo(repoFilters []string) bool {
 	return false
 }
 
+// A type that counts how many repos with a certain label were excluded from search results.
+type excludedRepos = struct {
```

Comment:
Declaring a struct type with `=` has weird subtle differences you probably do not want. I didn't realize this was possible until I saw it in your PR just now.

It declares the type equal to an anonymous struct: https://play.golang.org/p/tJo403zJIdG

```suggestion
type excludedRepos struct {
```

Reply:
this is probably a typo :⭕

Reply:
Oh, ahahaha, It's actually a [type alias](https://golang.org/doc/go1.9#language) [declaration](https://golang.org/ref/spec#Type_declarations) to an anonymous struct which I guess in the worst case could end up acting like a pseudo-generic type, interesting: https://play.golang.org/p/tT7FoHRsqMN

Reply:
Oh I'm familiar with the type alias syntax but haven't used it to directly alias based of a struct, that seems weird.

Reply:
Yeah it'd definitely be an abuse of the type-system to use it in that way ;) I just though it was funny

Reply:
[Some protobuf stuff I guess ¯\\\_(ツ)_/¯](https://sourcegraph.com/search?q=%2Ftype+%5Cw%2B+%3D+struct+%7B%2F+lang%3Ago+count%3A1000&patternType=regexp#36)

Reply:
Wow, of course it would be used there.. I guess someone _did_ have a real use for it